src/nvim/eval.c

```diff
@@ -10544,10 +10545,10 @@ typval_T eval_call_provider(char *provider, char *method, list_T *arguments, boo
   struct caller_scope saved_provider_caller_scope = provider_caller_scope;
   provider_caller_scope = (struct caller_scope) {
     .script_ctx = current_sctx,
-    .sourcing_name = sourcing_name,
+    .sourcing_name = (uint8_t *)sourcing_name,
```

Comment:
hmm, seems `caller_scope.sourcing_name` (and the other two) was changed in error to `uint8_t *` instead of `char *`. If we are adding casts, we can just as well change it to `char *` directly